File: dict_and_tuples/circle.py
# -*- coding: utf-8 -*-
"""
Created on Wed Sep  4 06:08:50 2024

@author: Vicky
"""

def circle(r):
    pie = 3.1415927
    area = pie*r*r
    circumference = 2*pie*r
    diameter = 2*r
    return round(area,2), round(circumference, 2), round(diameter, 2)

def Main():
    radius = float(input("what is the radius of your circle? "))
    area, circumference, diameter = circle(radius)
    print(f"area is {area}; \ncircumference is {circumference}; \nand diameter is {diameter}")

if __name__ =="__main__":
    Main()

File: dict_and_tuples/circle.java
import java.util.Scanner;
import java.lang.Math;
public class circle {
    private double radius;
    
    circle(){
        this.radius = 1;
    }
    
    circle(double radius){
        this.radius = radius;
    }
    
    public double area (){
        return (Math.round(Math.PI * this.radius * this.radius *100))/100.0;
    }
    
    public double circumference(){
        return (Math.round(2* Math.PI * this.radius * 100))/100.0;
    }
    
    public double diameter (){
        return (Math.round(2 * this.radius * 100))/100.0;
    }
    
    public double get_radius (){
        return Math.round(this.radius*100)/100.0;
    }
    
    public void set_radius(double radius){
        this.radius = radius;
    } 
    
    //test code
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        System.out.println("What is your circle's radius? ");
        double radius = scan.nextDouble();
        circle c = new circle(radius);
        System.out.println("area: " + c.area() + "\nCircumference: " + c.circumference() + "\nDiameter: " + c.diameter());
        
    }
}

